pkg/cmd/search/repos/repos.go

```diff
@@ -138,6 +138,9 @@ func reposRun(opts *ReposOptions) error {
 	if err != nil {
 		return err
 	}
+	if len(result.Items) == 0 {
```

Comment:
This should have an additional condition `opts.Exporter == nil` to avoid using the NoResultsError in case there was an exporter. For example, exporting to JSON should produce an empty array (`[]`), not an error.